plugins/win-dshow/virtualcam-module/placeholder.cpp

```diff
@@ -78,6 +79,45 @@ static void convert_placeholder(const uint8_t *rgb_in, int width, int height)
 	}
 }
 
+static bool load_placeholder_external()
+{
+	Status s;
+
+	wchar_t file[MAX_PATH] = {'\0'};
+	PWSTR pszPath = NULL;
+
+	HRESULT hr = SHGetKnownFolderPath(FOLDERID_RoamingAppData,
+					  KF_FLAG_DEFAULT, NULL, &pszPath);
```

Comment:
Wouldn't this read AppData all the time, including for portable OBS installations?  Wouldn't `obs_module_config_path` be more appropriate?

Reply:
This is inside the virtual camera DLL, so it doesn't have access to libobs. The DLL is unaware of the concept of "portable mode", so I think checking AppData is fine.

Reply:
In this case, wouldn't a system that only has a portable install _require_ that the user creates an AppData directory for obs-studio to use this custom placeholder instead of being able to utilize the portable config dir?

Reply:
Sorry, I don't understand the question about a portable install.

Reply:
On Windows, OBS can be run in "portable mode", meaning it stores all of its configuration data inside the application folder instead of in AppData.

https://obsproject.com/forum/resources/obs-and-obs-studio-portable-mode-on-windows.359/
https://github.com/obsproject/obs-studio/wiki/Launch-Parameters

Reply:
I've made the requested change to load_placeholder_internal() ... and load_placeholder_external() has been removed.

Reply:
I don't see how the virtualcam-module DLL could know that OBS is running in portable mode and look for a custom placeholder.png file in some other location.

Reply:
Is there a different place for the placeholder.png file that would address your concern about portable mode while still being a standard location for OBS plugin data on Windows?

Reply:
The only solutions I can think of without changing a bunch of includes are:
A. Store the current custom placeholder path in obs-virtualcam.txt
B. Store the current custom placeholder path in registry.
C. Use "custom_placeholder.png" in the plugin's data directory (`data/obs-plugins/win-dshow`) for the override.

A and B are pretty gross.  A would require not accidentally breaking any places that OBS already parses obs-virtualcam.txt.  C is the least offensive solution, in my opinion.  It would work in both normal and portable installations, and it shouldn't be overwritten by the OBS updater or installer.

Perhaps @notr1ch has another idea.

Reply:
How would I find that directory relative to the OBS install directory from inside the virtualcam-module DLL?

Reply:
Suggestion A requires using libobs to find the obs-virtualcam.txt file, so that's out. Suggestion C doesn't seem like a possibility, since there does not seem to be a way for the virtualcam-module DLL to know where or how OBS was installed. 

That leaves two options that I can see:

1. The implementation currently in the PR, which uses a standard OBS-on-Windows location for the file (like the MacOS implementation in #4232), but likely won't work for portable installs.
2. Suggestion B ... using Windows' registry.

Reply:
Suggestion C is easy, we already detect where the DLL is already in the code.

Reply:
I think you misunderstood what I wrote. Suggestion C doesn't have anything to do with finding where the DLL is.
  
Suggestion C requires the DLL to find where OBS is installed.

Reply:
As far as I can tell, the only way to make this work for a standard install and a portable install, is for the DLL to look for the path in the Registry.

Any suggestions on what Registry Key should be used?